### PR TITLE
Pathfinding fixes - bugs #208, #211

### PR DESCRIPTION
**Pull request by mrbermell**
solve #208:

> The pathfinding algorithm doesn't check for touchdown and keeps expanding. This leads to MOVE-actions that have paths that will never be completed.

and solve #211 

> During a MoveAction, I think you should only pickup the ball if you select the square with the ball in. i.e. the pathfinding should stop expanding its search if it picks up the ball. It currently doesn't.

### CODE REVIEW

**Comment by mrbermell** [comment]
@njustesen , please review and approve :)

File: botbowl/core/pathfinding/python_pathfinding.py [hunk 1]
```diff
@@ -309,6 +315,9 @@ def _expand(self, node: Node, target=None):
         if self.carries_ball and node.position.x == self.endzone_x:
             return
 
+        if (not self.carries_ball) and node.position == self.game.get_ball_position():
```
Comment (njustesen) [inline botbowl/core/pathfinding/python_pathfinding.py:318]: The ball can be in the air, e.g. during Blitz or Quick Snap! We should make a check for that.

**Review by njustesen — CHANGES_REQUESTED** [review]
Looks good. Just a small thing regarding balls in the air.

If you make a quick smoke test in the web app and the pathfinding works well there, I think it's good to go after the fix.

File: botbowl/core/pathfinding/python_pathfinding.py (hunk 1 above)
Comment (mrbermell) [reply botbowl/core/pathfinding/python_pathfinding.py]: Done.

**Comment by mrbermell** [comment]
Solved your review comment and smoke test in web app is is fine after playing a drive against myself.

**Review by njustesen — APPROVED** [review]

**Pull request merged**